### PR TITLE
implement `get_tmp_cache`

### PR DESCRIPTION
**Pull request by ranocha**
This function is required for some interoperability with Trixi.jl.

CC @jmbender

### CODE REVIEW

File: src/mprk.jl [hunk 1]
```diff
@@ -337,6 +337,8 @@ struct MPEConservativeCache{PType, uType, tabType, F} <: OrdinaryDiffEqMutableCa
     linsolve::F
 end
 
+get_tmp_cache(integrator, ::MPE, cache::OrdinaryDiffEqMutableCache) = (cache.σ,)
```
Comment (ranocha) [inline src/mprk.jl:340]: I have used the same signatures/specializations as used in OrdinaryDiffEq.jl.

**Comment by ranocha** [comment]
Do you understand the error below, @SKopecz?

```
┌ Error: failed to run `@example` block in src/index.md:173-178
│ ```@example SIR
│ using Plots
│ 
│ plot(sol, label = ["S" "I" "R"], legend=:right)
│ plot!(sol, idxs = ((t, S, I, R) -> (t, S + I + R), 0, 1, 2, 3), label = "S+I+R") #Plot S+I+R over time.
│ ```
│   exception =
│    Incorrect specification of `idxs`
```

I have not changed this part of the docs and it worked before...

File: test/runtests.jl [hunk 2]
```diff
@@ -1262,7 +1262,19 @@ end
                 for prob in (prob_default, prob_tridiagonal, prob_dense, prob_sparse,
                              prob_default2,
                              prob_tridiagonal2, prob_dense2, prob_sparse2)
-                    solve(prob, alg; dt, adaptive = false)
+                    sol1 = solve(prob, alg; dt, adaptive = false)
+
+                    # test get_tmp_cache and integrator interface - modifying
+                    # values from the cache should not changes the final results
```
Comment (JoshuaLampert) [inline test/runtests.jl:1268]:
```suggestion
                    # values from the cache should not change the final results
```

**Review by JoshuaLampert — COMMENTED** [review]
> Do you understand the error below

I'm not sure, but I could imagine this comes from https://github.com/SciML/SciMLBase.jl/pull/645.

**Comment by JoshuaLampert** [comment]
You could test that by forcing SciMLBase@2.44.0.

**Comment by ranocha** [comment]
Good idea. It's running in #114

**Comment by ranocha** [comment]
Yes, that's the issue: https://github.com/SciML/SciMLBase.jl/issues/746

**Comment by ranocha** [comment]
Re-running CI so that we can check the results with the latest version of SciML including a bug fix.

**Comment by ranocha** [comment]
No, didn't help...

**Comment by SKopecz** [comment]
I'd suggest to discard the line `plot!(sol, idxs = ((t, S, I, R) -> (t, S + I + R), 0, 1, 2, 3), label = "S+I+R")` in the docs. I find the syntax `idxs = ((t, S, I, R) -> (t, S + I + R), 0, 1, 2, 3)` rather cumbersome. Instead we could use something like
```julia
plot!(sol.t,sum.(sol.u), label = "S+I+R")
```

**Comment by ranocha** [comment]
> I'd suggest to discard the line `plot!(sol, idxs = ((t, S, I, R) -> (t, S + I + R), 0, 1, 2, 3), label = "S+I+R")` in the docs. I find the syntax `idxs = ((t, S, I, R) -> (t, S + I + R), 0, 1, 2, 3)` rather cumbersome. Instead we could use something like
> 
> ```julia
> plot!(sol.t,sum.(sol.u), label = "S+I+R")
> ```

Done in https://github.com/SKopecz/PositiveIntegrators.jl/pull/113/commits/787d8a5a73b7bf3e89b0b058d388090cddd94b9e

**Review by SKopecz — APPROVED** [review]

**Pull request merged**